cirq/google/engine/engine_job.py

```diff
@@ -285,6 +285,10 @@ def results(self) -> List[study.TrialResult]:
                   result_type == 'cirq.api.google.v2.Result'):
                 v2_parsed_result = v2.result_pb2.Result.FromString(result.value)
                 self._results = self._get_job_results_v2(v2_parsed_result)
+            elif result_type == 'cirq.google.api.v2.BatchResult':
+                v2_parsed_result = v2.batch_pb2.BatchResult.FromString(
+                    result.value)
+                self._results = self._get_batch_results_v2(v2_parsed_result)
```

Comment:
It seems rather odd to flatten this down to a single list of results across all programs in the batch. I would think we might want a new API to access batch results in a more structured manner.

Reply:
Yeah, my plan is to have something like that in a future PR that extends the interface.  This will fit into the current implementation, and then we can have a new API to grab batch results in a follow-up.  

Sound good?